consensus/bor/span_store.go

```diff
@@ -0,0 +1,63 @@
+package bor
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/consensus/bor/heimdall/span"
+	"github.com/ethereum/go-ethereum/log"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// SpanStore acts as a simple middleware to cache span data populated from heimdall. It is used
+// in multiple places of bor consensus for verification.
+type SpanStore struct {
+	store          *lru.ARCCache
+	heimdallClient IHeimdallClient
+}
+
+func NewSpanStore(heimdallClient IHeimdallClient) SpanStore {
+	cache, _ := lru.NewARC(10)
+	return SpanStore{
+		store:          cache,
+		heimdallClient: heimdallClient,
+	}
+}
+
+const (
+	defaultSpanLength = 6400 // Default span length i.e. number of bor blocks in a span
```

Comment:
Similar question for `defaultSpanLength`, could this be read from contract?

Reply:
Actually yes. We can fetch it from heimdall params. Will update the code to do so.

Reply:
Updated the logic in https://github.com/maticnetwork/bor/pull/1472/commits/fc2462d0ee52c0bbd1bd5a686c5c3873f24584c9 as discussed today to use existing span's start and end block (as done in contracts) instead of relying on this formula with hardcoded values.